### PR TITLE
Split cli.py into multiple smaller files

### PR DESCRIPTION
**Pull request by FrNecas**
First step towards #1082 (this hopefully shouldn't affect too many open PRs with merge conflicts).

I just split the file into a separate file for each subgroup of the main command which made the files IMO easier to navigate. One catch is that I had to make the "command tree" differently, by explicitly calling `main.add_command()`, just using decorators would not suffice (the code would not be run). To achieve this, I had to separate `CustomClass` into its own file to avoid circular imports (otherwise `__init__` would import a command, and command would try to import `CustomClass` from `__init__`)

### CODE REVIEW

**Comment by FrNecas** [comment]
I also wanted to remove the click-7 workarounds (the oldest fedora stable, F35, has click-8) but then realized that we still support epel8 :/

**Comment by happz** [comment]
FTR, one PR with merge conflicts would be https://github.com/teemtee/tmt/pull/1530 for sure (if that PR gets accepted...).

**Comment by FrNecas** [comment]
Sorry, missed that PR but I see that we do a similar thing :) Let's try to merge yours first, I will then rebase mine (will probably be easier that way). Looks like I still have some more work to do, looking at the tests.

**Comment by happz** [comment]
> Sorry, missed that PR but I see that we do a similar thing :) Let's try to merge yours first, I will then rebase mine (will probably be easier that way). Looks like I still have some more work to do, looking at the tests.

Np, that's fine. It's just an extension to some type of annotation work around `ClickDecoratorCallback` & co. I didn't plan to split `cli.py` at all, just streamline options a bit. It's not a critical patch, therefore I didn't push it forward as I did with many others, still it'd be nice to have it merged, definitely.

**Comment by FrNecas** [comment]
Hm, I 'm just thinking what is better when we consider #1530 :+1: 

a) keep `cli/common_options` where we initialize the option decorators (just using the more generalized way instead of the current duplication)
b) move the decorator initialization to `options.py`

What do you think?

**Comment by FrNecas** [comment]
Hm, internal plugin detection seems to work correctly even with the new changes but a test weirdly fails

File: tmt/cli/__init__.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+from tmt.cli.click_group import CustomGroup
+from tmt.cli.common_options import verbosity_options
+from tmt.cli.init import init
+from tmt.cli.lint import lint
+from tmt.cli.plans import plans
+from tmt.cli.run import run
+from tmt.cli.setup import setup
+from tmt.cli.status import status
+from tmt.cli.stories import stories
+from tmt.cli.tests import tests
```
Comment (happz) [inline tmt/cli/__init__.py:23]: This global namespace pollution does not seem right to me. The only purpose of `tests` is to be used in `main.add_command(tests)` call at the end of the file. Could we use some kind of import magic?

```python
for name in ('run', 'tests', 'plans', ...):
  command = tmt.plugins.import_member(f'tmt.cli.{name}, name)
  main.add_command(command)
```

**Comment by happz** [comment]
It seems to me things are living in the wrong places - root command in `__init__.py`, commands and their support classes and variables in other modules. How about changing the locations a bit?

* move the root command, `main`, into its own module, e.g. `root.py` or `main.py`,
* move support code - context, group, options - to `__init__.py`,
* change `tmt` to run `tmt.cli.main.main()`.

* command would import support code from `tmt.cli` aka `tmt/cli/__init__.py`
* *all* commands would live in `tmt.cli` submodules
* `tmt.cli.main` is free to import other commands, there should be no trap or loop in the import sequence `tmt.cli.main` -> `tmt.cli.tests` -> `tmt.cli`.

Custom classes for CLI implementation in submodules to break import loops do not sound right :/ If I start working on a custom tmt subcommand, I'd expect all building blocks to be available from `tmt.cli`, not 3 distinct submodules `tmt.cli.{click_context_object,click_group,common_options}`.

I would maybe even consider renaming command modules, `tests.py` => `cmd_tests.py`, in a fashion similar to `bkr`, to help with identifying what all those files are about.

File: tmt/cli/__init__.py (hunk 1 above)
Comment (FrNecas) [reply tmt/cli/__init__.py:23]: I guess we could but I personally don't see nothing wrong with being explicit in importing. If You don't like potential name clashes, we could use absolute imports (but I prefer `from` imports but that's a matter of personal taste)

**Comment by FrNecas** [comment]
>  move the root command, main, into its own module, e.g. root.py or main.py

That's what I considered as well. The problem with this is that it breaks the API. It would be easy to fix in just this repo but the `tmt.cli.main` "main entrypoint" is imported also in tests in the internal plugins repo. So this leads to a chicken and egg problem - we would like to change the import here but can't because it would break downstream tests; and we can't change downstream tests because we don't have the changes here :)

**Comment by happz** [comment]
> > move the root command, main, into its own module, e.g. root.py or main.py
> 
> That's what I considered as well. The problem with this is that it breaks the API. It would be easy to fix in just this repo but the `tmt.cli.main` "main entrypoint" is imported also in tests in the internal plugins repo. So this leads to a chicken and egg problem - we would like to change the import here but can't because it would break downstream tests; and we can't change downstream tests because we don't have the changes here :)

I wouldn't block on this, because we've done that several times already :) Use the `breakage` label, prepare downstream patches & sync on IRC :)

**Comment by psss** [comment]
Ongoing in different pull requests, step by step. Rebase here would be painful.

**Pull request closed without merging**